### ISSUE TITLE
Quarto errors when comment containing #| echo:false

### ISSUE THREAD

**Issue by dlakelan** [quarto-dev/quarto-cli#6344]
### Bug description

in a julia code block with the following code, where a space is missing after the colon 

```{julia}
#| echo:false
...some code
```

quarto render exits with:

```
ERROR: TypeError: Cannot delete property '0' of [object String]

TypeError: Cannot delete property '0' of [object String]
    at file:///opt/quarto/bin/quarto.js:53103:46
    at Array.forEach (<anonymous>)
    at flushLineBuffer (file:///opt/quarto/bin/quarto.js:53094:30)
    at quartoMdToJupyter (file:///opt/quarto/bin/quarto.js:53142:17)
    at async createNotebookforTarget (file:///opt/quarto/bin/quarto.js:55428:16)
    at async Object.target (file:///opt/quarto/bin/quarto.js:55221:18)
    at async fileExecutionEngineAndTarget (file:///opt/quarto/bin/quarto.js:58033:20)
    at async renderContexts (file:///opt/quarto/bin/quarto.js:82635:34)
    at async Object.renderFormats (file:///opt/quarto/bin/quarto.js:82688:26)
    at async inputTargetIndex (file:///opt/quarto/bin/quarto.js:83504:21)
```

if you replace it with:

```
#| echo: false
```
including the space, it will render properly.

### Expected behavior

either render the document, or give a meaningful error that would lead the user to fix the problem with the missing space

### Your environment

Debian GNU/Linux with current quarto deb file as of 2023-07-26

**Comment by mcanouil** [comment]
Thanks for the report!

Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by dlakelan** [comment]
````qmd
```{julia}

#| echo:false

1+1
```
````

**Comment by mcanouil** [comment]
We need a complete document as in my example.

**Comment by cderv** [comment]
I think the example shared is enough. I can reproduce this error with Julia but also `python` cells. 

You have this error on two reason : 

* `echo:false` is not valid YAML because you are missing the space. Without the space, YAML parsing can't know this is `key: value`
* The space at the top of the cells prevents you to see the warning you would have in IDE like VSCODE. Without the blank line, you would have see that warning

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/18dce717-b6fe-48e6-a077-7958790695dc)

This YAML linting is part our way to give a meaningful error that would lead the user to fix the problem with the missing space. 

So basically, don't start your code cells with a blank line 

@cscheid can we do more with YAML validation in this case to error like with a wrong value for an option ?

**Comment by dlakelan** [comment]
In VSCodium using Quarto 1.90.1 extension I don't see any linting when I put 

#| echo:false 

as the first line of the code block

**Comment by mcanouil** [comment]
From the latest version of VSCode and the Quarto extension (and development version of Quarto CLI):
<img width="820" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/7eb7b6a7-ea07-4124-b42c-1bdba3133f09">

**Comment by cscheid** [comment]
I can't reproduce:

<img width="411" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/53ea2c02-b8a3-4656-8cbe-c274318a7cb8">

<img width="734" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/7b30abdb-5141-438c-aba9-4fe5303a9007">


Did you try saving your file? Our linting doesn't kick in until you save the file.

**Comment by dlakelan** [comment]
For some reason it wasn't working even after saving, but then after doing some other stuff it is working at the moment. This may have been unrelated and due to some issues with an NFS server acting slightly wonky.

**Comment by cscheid** [comment]
Ok, so there are two problems here.

First, we do need to fix our code somehow to avoid that error message. However, even if we do, quarto currently doesn't recognize metadata that doesn't start _right at the beginning of the document_. This is good syntax:

````
```{python}
#| echo: false
...
```
````

This is bad syntax:
````
```{python}

#| echo: false
...
```
````

Specifically, the former syntax triggers our yaml validator, but the latter syntax does not. Unfortunately, it seems that the latter syntax is still recognized when we attempt to run the python cell. The error is slightly better, but not great, because it's past our validator:

````
ERROR:

None of the known patterns match for echo:false
````

This points to an inconsistency between the jupyter and knitr engines, and we can't change one without changing the other. I'm not sure how to proceed.

**Comment by cscheid** [comment]
I'm closing this as best we can right now; we can't do much better because the error is no longer happening in quarto, but in `poyo`, the Python yaml parsing library we use. In the future we want to be able to do better by having a linter point to the  whitespace issue, (which would lead to the yaml validation message).
